app/controllers/prizes_controller.rb

```diff
@@ -25,6 +29,16 @@ def win_prize
 
   private
 
+  def sum_prizes_percentages
+    total_percentage = 0
+
+    Prize.all.each_with_object({}) do |prize, _acc|
+      total_percentage += prize.percentage
+    end
+
+    total_percentage
+  end
+
```

Comment:
There's a one line solution for this:

```
Prize.pluck(:percentage).sum
```

`pluck` is a method that retrieves all the values from the column/attribute we pass as parameter.
`Prize.pluck(:percentage)` will retrieve an Array with the percentage values for all the Prizes. Since we are simply adding all the percentages, we can call the `sum` method of the Array class to get the final sum

Reply:
Nothing like a one-liner to replace 10 lines. So fresh and so clean clean. 🧼

Reply:
Thanks, it works!